### DISCUSSION TITLE
Imagemagick displaying a single page tiff/tif file as two pages in MediaWiki

### DISCUSSION THREAD

**Discussion by gtaillefer** [ImageMagick/ImageMagick#5632] [Help]
Hello, I'm trying to use an extension on MediaWiki called PagedTiffHandler which uses ImageMagick to display tiff files on MediaWiki. My specifications are as such:
- Bluehost/cpanel shared hosting
- MediaWiki 1.38.1
- Extension: [PagedTiffHandler](https://www.mediawiki.org/wiki/Extension:PagedTiffHandler)
- CentOS 6
The problem that I'm having is that whenever I upload a tiff/tif file, I get one of three things. The most common situation is where when I go to see the tiff file on its page, even though that the file is only one page, it displays it as two. The first one with a grey background with an error message saying: `Error creating thumbnail: File with dimensions greater than _#_ MP`, with the next page being just the actual file itself. The second situation which has only occurred once, is the exact same situation above, but both the first and second page show the same thing just fine, no error messages. Then the third situation which has only happened once, is where the tiff file uploads, there isn't two pages, and the tiff file shows just fine without any error messages. 
[I had asked about this issue on MediaWiki](https://www.mediawiki.org/wiki/Topic:X4gq3p8auavyylk6), and one of the developers told me that this was an ImageMagick issue. Any help would be welcome, thanks

**Comment by snibgo** [comment]
What version of IM? Version 7.1.0-47 had problems reading TIFF files, and I suggest it is NOT used for that purpose. Instead, use 7.1.0-49 or later.

Aside from that, to diagnose the problem, we need to reproduce it. Ideally you can show a sample input and output, and the IM command that made that output.

Obviously, if the input TIFF contains two images and a bracketed number is not used (eg "`infile.tiff[0]`") then the output will contain two images.

I followed the link to https://www.mediawiki.org/wiki/Special:ExtensionDistributor?extdistname=PagedTiffHandler&extdistversion=REL1_38 but the tar.gz contains a load of stuff. I looked at PagedTiffHandler, which builds ImageMagick commands, but there are too many variables for me to guess what the actual command would be.

**Reply by gtaillefer** [reply]
I checked the version and it responded with: `Version: ImageMagick 6.9.10-68 Q16 x86_64 2021-10-14 https://imagemagick.org`, yeah not that great, I'll look into trying to get an updated version, but with shared hosting that will be pretty difficult to not be able to do something because it is blocked. 
As for the reproducing thing, if you need the files here are examples of the first problem: 
https://wwii-archives.org/index.php?title=File%3AHhhhhhhhh.tif&page=1
https://wwii-archives.org/index.php?title=File%3ASometiff.tif&page=1
https://wwii-archives.org/index.php?title=File%3AA_letter.tif&page=1

Heres the second problem:
https://wwii-archives.org/index.php?title=File:Untitled_(4).tif&page=1

And heres the third: 
https://wwii-archives.org/index.php/File:Yes.tiff

I'm not really sure about the specifics on PagedTiffHandler's code, so I can't really say much about that.

**Reply by fmw42** [reply]
How about posting the actual tif file!

**Comment by snibgo** [comment]
For the first situation "Hhhhhhhhh.tif", the link says:
```
Original file ‎(8,738 × 13,457 pixels, file size: 340.82 MB, MIME type: image/tiff, 2 pages) 
```
So it has two pages, ie two images. A simple "`convert`" command will give two output images. The link at "Original file" downloads a file that has zero bytes.

For the second situation "Untitled_(4).tif", the download works.
```
f:\web\im>%IMG7%magick "Untitled_(4).tif" info:
Untitled_(4).tif[0] TIFF 8913x5818 8913x5818+0+0 8-bit TrueColor sRGB 152.701MiB
 0.000u 0:00.000
Untitled_(4).tif[1] TIFF 1515x989 1515x989+0+0 8-bit TrueColor sRGB 0.016u 0:00.
016
```
So that file certainly has two images. As you say you have two outputs, there is no problem.

For the third situation, there is no problem. 

If you have an input TIFF with two images and you want to create a single PNG output from the first image:
```
magick in.tiff[0] out.png
```
... or for IM v6:
```
convert in.tiff[0] out.png
```

**Reply by gtaillefer** [reply]
Sorry my response has taken a while, could you lead me to a place were I could find how to update my IM to 7.1.0-49 or later? As you can see I'm on shared hosting but when I look that up on, how to update ImageMagick on shared hosting, the results come with "how to _install_ ImageMagick on shared hosting". Thanks

**Comment by fmw42** [comment]
As far as I know, you cannot update the installation of Imagemagick on the shared host.  They have to do it for the version they provide.  But some ISPs will allow you to install your own version of Imagemagick in your own directory structure. You will need to contact your ISP about how to do that or if they have to do it for you.

**Reply by gtaillefer** [reply]
Well the problem with that is that I had just asked on Bluehost support if they would allow me to run "yum install epel-release" which they said wasn't allowed because I was on shared hosting. If they weren't able to run that for me, I'm not sure if they would allow me to do as you mentioned. So I guess I'm stuck with this problem unfortunately.
